File: frontend/src/domain/entities/jobPosition.ts
export interface JobPosition {
  id: string;
  company_id: string;
  name: string;
  description: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface JobPositionCreate {
  company_id: string;
  name: string;
  description?: string;
  is_active?: boolean;
}

export interface JobPositionUpdate {
  name?: string;
  description?: string;
  is_active?: boolean;
}

File: backend/app/application/dto/job_position.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class JobPositionCreate(BaseModel):
    company_id: UUID
    name: str
    description: str | None = None
    is_active: bool = True


class JobPositionUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    is_active: bool | None = None


class JobPositionResponse(BaseModel):
    id: UUID
    company_id: UUID
    name: str
    description: str | None
    is_active: bool
    created_at: datetime
    updated_at: datetime

